ibis/backends/tests/test_interactive.py

```diff
@@ -105,3 +105,18 @@ def test_no_recursion_error(con, monkeypatch):
         exc.RelationError, match="The scalar expression cannot be converted"
     ):
         repr(expr)
+
+
+def test_scalar_uses_pyarrow(con, table, monkeypatch, mocker):
+    monkeypatch.setattr(ibis.options, "interactive", True)
+
+    execute_spy = mocker.spy(con, "execute")
+    to_pyarrow_spy = mocker.spy(con, "to_pyarrow")
+
+    repr(table.limit(1).string_col)
+
+    # execute doesn't get called
+    execute_spy.assert_not_called()
```

Comment:
> Right, we have some backends that actually go through pandas to get to pyarrow.

If things pass otherwise, the fix seems fine to me without this test at all. No need to assert we're not calling `execute` when reading the formatting code it's clear we go through `to_pyarrow`.

Reply:
The main point of this test was to make it explicit what happens and in this case make us aware of any backends that have coupling between the pandas and pyarrow format converters. I think I'll leave the test, with appropriate xfails, as a form of documentation.